src/Sylius/Bundle/ApiBundle/Resources/config/validation/AddPaymentRequest.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+
+ This file is part of the Sylius package.
+
+ (c) Sylius Sp. z o.o.
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping http://symfony.com/schema/dic/services/constraint-mapping-1.0.xsd">
+    <class name="Sylius\Bundle\ApiBundle\Command\Payment\AddPaymentRequest">
```

Comment:
Guessing some validation of whether the `type` passed is even covered in the selected payment method would be nice, then we can return a list of available request types for each gateway.

Reply:
I though about this exact same idea yesterday 😄

Reply:
I made a validator : `ChosenPaymentRequestActionEligibilityValidator` by enhancing the service in charge of providing CommandProviders.